svgcheck/run.py

```diff
@@ -29,6 +29,11 @@ def clear_cache(cache_path):
     sys.exit()
 
 
+def delete_tempfile(tempfile_in_use, source):
```

Comment:
Consider:
```
>>> import tempfile
>>> f = tempfile.NamedTemporaryFile()
>>> type(f)
<class 'tempfile._TemporaryFileWrapper'>
>>> from tempfile import _TemporaryFileWrapper
>>> isinstance(f,_TemporaryFileWrapper)
True
```
It does dig for an internal, but it would be less likely to accidentally remove a file that _wasn't_ a temp file because of an error in managing the boolean.

Reply:
For the current pattern, I'd keep it simpler - do away with the boolean flag and just keep the name of the file to clean up in `tempfile_name`. If that's `None`, there's no cleanup.

I think there's a cleaner way, though, I'll suggest separately

Reply:
I refactored the code, so tempfile library will handle the deletion of the temporary files.
This library has significant improvement from Python 3.12 onwards.
Because of the same reason, `svgcheck` will only support stdin on Windows with Python >= 3.12.